base/Hero.ts

```diff
@@ -61,6 +62,7 @@ export class Hero extends ControllableChar {
 
     private arrow_inputs: number;
     private force_diagonal_speed: {x: number; y: number} = {x: 0, y: 0};
+    private stick_dashing: boolean = false;
```

Comment:
I'm not in favor to have a `stick_dashing` since we don't have it in the original game, but if you really want to keep it, you have to make it independent from the type of input hardware you have. The engine should see only GBA controls or controls to debug the engine.